src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Controllers/AuthenticationController.cs

```diff
@@ -256,6 +281,67 @@ public async Task<ActionResult> RefreshJwtCookie()
             }
         }
 
+        /// <summary>
+        /// Action for converting a JWT generated by <c>ID-porten</c> with a new JWT for further use as authentication against rest of Altinn.
+        /// </summary>
+        /// <returns>The result of the action. Contains the new token if the old token was valid and could be converted.</returns>
+        private async Task<ActionResult> AuthenticateIdPortenToken(string originalToken)
+        {
+            try
+            {
+                ICollection<SecurityKey> signingKeys =
+                   await _signingKeysRetriever.GetSigningKeys(_generalSettings.IdPortenWellKnownConfigEndpoint);
+
+                TokenValidationParameters validationParameters = new TokenValidationParameters
+                {
+                    ValidateIssuerSigningKey = true,
+                    IssuerSigningKeys = signingKeys,
+                    ValidateIssuer = false,
+                    ValidateAudience = false,
+                    RequireExpirationTime = true,
+                    ValidateLifetime = true
+                };
+
+                _validator.ValidateToken(originalToken, validationParameters, out _);
+                _logger.LogInformation("Token is valid");
+
+                JwtSecurityToken token = _validator.ReadJwtToken(originalToken);
+
+                string pid = token.Claims.Where(c => c.Type.Equals(PidClaimName)).Select(c => c.Value).FirstOrDefault();
+                string authLevel = token.Claims.Where(c => c.Type.Equals(AuthLevelClaimName)).Select(c => c.Value).FirstOrDefault();
+                string authMethod = token.Claims.Where(c => c.Type.Equals(AuthMethodClaimName)).Select(c => c.Value).FirstOrDefault();
+
+                if (string.IsNullOrWhiteSpace(pid) || string.IsNullOrWhiteSpace(authLevel) || string.IsNullOrWhiteSpace(authMethod))
+                {
+                    _logger.LogInformation("Token containted invalid or missing claims.");
+                    return Unauthorized();
+                }
+
+                UserProfile userProfile = await _userProfileService.GetUser(pid);
+
+                List<Claim> claims = new List<Claim>();
```

Comment:
I belive in this scenario we can add PID. We are only converting to another token (not cookie). 
So we could keep it. Do you agree @alt-how

Reply:
@TheTechArch  I was considering keeping all the original claims from id-porten, but didn't see how we would utilize them in our own solution? DO you have any use cases where it would be beneficial to keep the claims?

Reply:
@TheTechArch I did consider using exchange when I created the original endpoint, but exchange is a term in oAuth 2.0. I didn't want to create an expectation that we had functionality related to Token Exchange. What we are making does not confer with the standard as far as I know.

Reply:
@TheTechArch I'm happy as long as we are not storing pid in local storage

Reply:
@acn-sbuad  in the sirius use case the app will forward the token to the Sirius API.  Without the PID they would not know who is authenticated

Reply:
Ah, okay.  I'll add all tokens from id-porten and remove duplicate keys then before adding ours.

Reply:
@TheTechArch  I've updated the claims and now it would look something like this. Does that seem okay? 

```json
{
  "nameid": "20000011",
  "urn:altinn:userid": "20000011",
  "urn:altinn:username": "",
  "urn:altinn:partyid": 50002119,
  "urn:altinn:authenticatemethod": "Minid-PIN",
  "urn:altinn:authlevel": 3,
  "sub": "kBSMtI7PCrNRH9jlKtVNpj9upKf-S1w1M9zgCIxlz0A=",
  "amr": "Minid-PIN",
  "pid": "111077XXXXX",
  "locale": "nb",
  "nonce": "1585045781364132",
  "sid": "BYSqEpVGRrh6rElmnzzTjcU0roC95rxNCC2kAsB2hmY",
  "acr": "Level3",
  "auth_time": 1585045793,
  "exp": 1585047785,
  "iat": 1585045985,
  "nbf": 1585045985
}
```

Reply:
@TheTechArch I probably remember wrong, but shouldn't nameid and sub have the same value?

Reply:
we should probably remove the sub-claim then. But based on the description in id-portens documentation I din't see how it conflicted with our nameid value. 

> “subject identifier” - an unique identifier for the authenticated user. The value is pairwise, meaning a given client will always get the same value, whilst different clients do not get equal values for the same user.

Reply:
We use sub when login in. We user userID as the value.  Look at login API for cookie.

Reply:
So should the sub-value generated by id-porten be ignored or included in the new token?

Reply:
removing the sub claim for generated token.